File: cpp/leetcode/170.cpp
class TwoSum {
    unordered_map<int,int> map;
public:
    void add(int number) {
        map[number]++;
    }

    bool find(int value) {
        for (auto it : map) {
            int i = it.first;
            int j = value - i;
            if ((i == j && it.second > 1) || (i != j && map.find(j) != map.end())) {
                return true;
            }
        }
        return false;
    }
};

File: python/leetcode/170.py
class TwoSum:
    def __init__(self):
        self.ctr = {}

    def add(self, num):
        self.ctr[num] = self.ctr.get(num, 0) + 1

    def find(self, value):
        ctr = self.ctr
        for num in ctr:
            if value - num in ctr and (value - num != num or ctr[num] > 1):
                return True
        return False